core/BaseComponent.js

```diff
@@ -404,15 +420,61 @@ export class BaseComponent extends HTMLElement {
     return this.__cssDataCache[propName];
   }
 
+  /**
+   * @private
+   * @param {String} ctxPropName
+   */
+  __extractCssName(ctxPropName) {
+    return ctxPropName
+      .split('--')
+      .map((part, idx) => {
+        return idx === 0 ? '' : part;
+      })
+      .join('--');
+  }
+
+  updateCssData = () => {
+    this.dropCssDataCache();
+    this.__boundCssProps?.forEach((ctxProp) => {
+      let val = this.getCssData(this.__extractCssName(ctxProp), true);
+      this.$[ctxProp] !== val && (this.$[ctxProp] = val);
+    });
+  };
+
+  /** @private */
+  __initStyleAttrObserver() {
+    if (!styleMutationObserverCbList) {
+      styleMutationObserverCbList = new Set();
+    }
+    styleMutationObserverCbList.add(this.updateCssData);
+    if (!styleMutationObserver) {
+      styleMutationObserver = new MutationObserver((/** @type {MutationRecord[]} */ records) => {
+        records[0].type === 'attributes' &&
+          styleMutationObserverCbList.forEach((cb) => {
+            cb();
```

Comment:
I have an idea how to reduce callback calls. We could detect if `style` mutation can affect our state - check the new `style` value for the occurrences of any known css properties (or just any css property at least)

Like

1. Got mutation `margin: 0; --cfg-multiple: 0;` -> `margin: 0; --cfg-multiple: 1;`
2. Ok, we have `--cfg-multiple`
3. Run state update callback

Reply:
yep